File: FoxChat-vue/src/api/llmConfig.js
import request from '@/utils/request';

/**
 * LLM 配置 API
 * 提供创造物的模型配置接口
 */

// 批量保存配置（5 个场景）
export function saveConfigsBatch(llmId, configs) {
  return request.post('/llm/config/batch', {
    llmId,
    configs
  });
}

// 获取所有配置
export function getConfigs(llmId) {
  return request.get(`/llm/config/${llmId}`);
}

// 删除配置
export function deleteConfig(llmId, scenario) {
  return request.delete(`/llm/config/${llmId}/${scenario}`);
}

// 测试连接
export function testConnection(config) {
  return request.post('/llm/config/testConnection', {
    modelName: config.modelName,
    apiKey: config.apiKey,
    baseUrl: config.baseUrl,
    scenario: config.scenario
  });
}

// 验证配置完整性
export function validateConfig(llmId) {
  return request.get(`/llm/config/validate/${llmId}`);
}

File: FoxChatRAG-python/app/api/llm_config.py
"""
LLM 配置 API 路由

端点:
- POST /llm/testConnection: 测试 LLM 连接（被 Java Feign 调用）

注意：
  LLM 配置的 CRUD（增删改查）由 Java 端 LlmConfigController + LlmConfigServiceImpl 直接操作 MySQL，
  Python 端不再暴露重复的 CRUD 端点。
  仅保留 testConnection，因为只有 Python 安装了 litellm 可以真正发起 API 连通性测试。
"""

from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.db.mysql_client import get_db
from app.schemas.M import M
from loguru import logger

llm_config_router = APIRouter(prefix="/llm", tags=["llm-config"])


@llm_config_router.post("/testConnection")
async def test_connection(config_data: dict, db: AsyncSession = Depends(get_db)):
    """
    测试 LLM 连接（被 Java Feign ChatClient.testConnection 调用）

    Args:
        config_data: 配置数据 (model_name, api_key, base_url)
        db: 数据库会话

    Returns:
        测试结果 (成功/失败, 错误消息)
    """
    from app.service.llm_config_service import test_llm_connection

    logger.info(f"【API】测试连接: model={config_data.get('model_name')}")

    result = await test_llm_connection(
        model_name=config_data["model_name"],
        api_key=config_data["api_key"],
        base_url=config_data["base_url"]
    )

    return M.get_msg(result)